internal/DockerMemoryLimit.go

```diff
@@ -0,0 +1,188 @@
+package internal
+
+import (
+	"errors"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+//DockerMemoryLimit -
+type DockerMemoryLimit struct {
+	_initSchedFile          string
+	_selfCpusetFile         string
+	_memoryCgroupPrefixPath string
+	_memLimitInBytesFile    string
+	readDone                bool
+	readlock                sync.Mutex
+	isDocker                bool
+	containerID             string
+	containerIDError        error
+	memLimitInBytes         uint64
+	memLimitInBytesError    error
+}
+
+func NewDockerMemoryLimit() (d DockerMemoryLimit) {
+	d._initSchedFile = "/proc/1/sched"
+	d._selfCpusetFile = "/proc/self/cpuset"
+
+	//Example:
+	//  Full Path
+	//  /sys/fs/cgroup/memory/system.slice/docker-779f13691a49de55b57928e8b8ed616a89388cc517c14a4ca955e2484c3fa632.scope/memory.limit_in_bytes
+	//  ContainerID
+	//  /system.slice/docker-779f13691a49de55b57928e8b8ed616a89388cc517c14a4ca955e2484c3fa632.scope
+	//  Formula
+	//  /sys/fs/cgroup/memory{containerID}/memory.limit_in_bytes
+	d._memoryCgroupPrefixPath = "/sys/fs/cgroup/memory"
+	d._memLimitInBytesFile = "/memory.limit_in_bytes"
+	return
+}
+
+func (d *DockerMemoryLimit) init() {
+	if !d.readDone {
+		d.readlock.Lock()
+		defer d.readlock.Unlock()
+		if !d.readDone {
+			d.readIsDocker()
```

Comment:
it shouldn't matter that we are in docker, we should use memory cgroup limit if there is one